src/overlays/actors/ovl_En_Trt/z_en_trt.c

```diff
@@ -22,176 +99,1708 @@ const ActorInit En_Trt_InitVars = {
+}
+
+void EnTrt_SetupLookToShopkeeperFromShelf(GlobalContext* globalCtx, EnTrt* this) {
+    play_sound(NA_SE_SY_CURSOR);
+    this->drawCursor = 0;
+    this->actionFunc = EnTrt_LookToShopkeeperFromShelf;
+}
+
+void EnTrt_LookToShopkeeperFromShelf(EnTrt* this, GlobalContext* globalCtx) {
+    if (this->cutsceneState == ENTRT_CUTSCENESTATE_PLAYING) {
+        ActorCutscene_Stop(this->cutscene);
+        this->cutsceneState = ENTRT_CUTSCENESTATE_STOPPED;
+    }
+    if (this->cutsceneState == ENTRT_CUTSCENESTATE_STOPPED) {
+        if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+            ActorCutscene_Stop(0x7C);
+        }
+        this->cutscene = this->lookToShopkeeperCutscene;
+        ActorCutscene_SetIntentToPlay(this->cutscene);
+        this->cutsceneState = ENTRT_CUTSCENESTATE_WAITING;
+    } else if (this->cutsceneState == ENTRT_CUTSCENESTATE_WAITING) {
+        if (ActorCutscene_GetCanPlayNext(this->cutscene)) {
+            ActorCutscene_StartAndSetFlag(this->cutscene, &this->actor);
+            this->cutsceneState = ENTRT_CUTSCENESTATE_PLAYING;
+            EnTrt_StartShopping(globalCtx, this);
+        } else {
+            ActorCutscene_SetIntentToPlay(this->cutscene);
+        }
+    }
+}
+
+void EnTrt_InitShopkeeper(EnTrt* this, GlobalContext* globalCtx) {
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600FEF0, &D_0600FD34, NULL, NULL, 0);
+    if (!(gSaveContext.weekEventReg[0x0C] & 8) && !(gSaveContext.weekEventReg[0x54] & 0x40) && gSaveContext.day >= 2) {
+        this->actor.draw = NULL;
+    } else {
+        this->actor.draw = EnTrt_Draw;
+    }
+}
+
+void EnTrt_InitShop(EnTrt* this, GlobalContext* globalCtx) {
+    u32 maxcolor = 0xFF;
+    EnTrt* this2;
+
+    EnTrt_GetCutscenes(this, globalCtx);
+    this->cutscene = this->lookForwardCutscene;
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 20.0f);
+    EnTrt_InitShopkeeper(this, globalCtx);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->actor.colChkInfo.cylRadius = 50;
+    this->timer = Rand_S16Offset(40, 20);
+    if (!(gSaveContext.weekEventReg[0xC] & 8) && !(gSaveContext.weekEventReg[0x54] & 0x40) && gSaveContext.day >= 2) {
+        this->textId = 0x84A;
+        this->actionFunc = EnTrt_ShopkeeperGone;
+    } else {
+        this->sleepSoundTimer = 10;
+        this->actor.textId = 0;
+        this->actionFunc = EnTrt_IdleSleeping;
+    }
+
+    this->stickAccumY = 0;
+    this->stickAccumX = 0;
+    this->cursorIdx = 0;
+    this->cursorY = this->cursorX = 100.0f;
+    this->cursorZ = 1.2f;
+    this->cursorColorR = 0;
+    this->cursorColorG = 80;
+    this->cursorColorB = maxcolor;
+    this->cursorColorA = maxcolor;
+    this->cursorAnimTween = 0.0f;
+    this->cursorAnimState = 0;
+    this->drawCursor = 0;
+
+    this2 = this;
+
+    this->stickLeftPrompt.stickColorR = 200;
+    this2->stickLeftPrompt.stickColorG = 200;
+    this2->stickLeftPrompt.stickColorB = 200;
+    this2->stickLeftPrompt.stickColorA = 180;
```

Comment:
More of an open question. These colours are defined as u32 right? I wonder if we should define a Color_RGBAu32 in `color.h` and update the `StickDirectionPrompt` struct from `z_en_girla.h`? There's something like that done for floats (`Color_RGBAf`), thoughts?

Reply:
I really like the idea. However since it affects more then just this actor (1 of which is still in PR) and would take some time to implement, I would prefer to do this in a separate PR after the NBS gets merged in.